lib/workspace.js

```diff
@@ -1,44 +1,38 @@
+const ERR = require('async-stacktrace');
 const request = require('request');
 
 const config = require('./config.js');
 const logger = require('./logger');
 
 module.exports = {
-
-    async controlContainer(callback, workspace_id, action) {
-        let promise = new Promise(resolve => {
-            request.post(`http://${config.workspaceContainerLocalhost}:${config.workspaceContainerPort}/`, {
-                json: {
-                    workspace_id: workspace_id,
-                    action: action,
-                },
-            }, (error, response, body) => {
-                if (error) {
-                    logger.error(`controlContainer error: ${error}`);
-                    return;
+    controlContainer(workspace_id, action, callback) {
+        const options = {
+            json: {
+                workspace_id: workspace_id,
+                action: action,
+            },
+        };
+        request.post(`http://${config.workspaceContainerLocalhost}:${config.workspaceContainerPort}/`, options, (err, res, body) => {
+            if (ERR(err, callback)) return;
+            logger.info(`controlContainer ${action} statusCode: ${res.statusCode}`);
+            logger.info(`controlContainer body: ${body}`);
+            if (res.statusCode == 200) {
+                if (action == 'init') {
+                    logger.info('controlContainer: container started');
                 }
-                logger.info(`controlContainer ${action} statusCode: ${response.statusCode}`);
-                logger.info(`controlContainer body: ${body}`);
-                if (response.statusCode == 200) {
-                    resolve(true);
-                } else {
-                    /* Display an error if we have one from the server */
-                    let server_error = 'unknown error';
-                    if (body) {
-                        server_error = body;
+                callback(null);
+            } else {
+                let server_error = 'unknown error';
+                if ('json' in body) {
+                    if ('data' in body.json) {
+                        // need to decode the `data` array to be human readable
+                        server_error = new Buffer.from(body.json.data).toString();
+                    } else if ('message' in body.json) {
+                        server_error = body.json.message;
                     }
-                    logger.error(`controlContainer could not connect to workspace host: ${server_error}`);
```

Comment:
Not sure if this was reverted intentionally or accidentally. This was @nicknytko's solution for making the json message human-readable.